lang/python/wiredtiger.i

```diff
@@ -668,6 +670,16 @@ OVERRIDE_METHOD(__wt_cursor, WT_CURSOR, search_near, (self))
 	}
 }
 
+%typemap(argout)(char **key_datap, int *key_sizep, char **value_datap, int *value_sizep) (
+    PyObject *key_data, PyObject *value_data) {
+	if (*$1 && *$3) {
```

Comment:
Could this be unwound, so that it supports either key or value being None (Python equivalent)?

i.e:
```
if (*$1)
  key_data = PyBytes_FromStringAndSize(*$1, *$2)
if (*$2)
    value_data = PyBytes_FromStringAndSize(*$3, *$4);
$result = SWIG_Python_AppendOutput(key_data, value_data);
```

Reply:
FWIW: I understand that the Python API doesn't really give users a mechanism for describing that (I don't think it's legal to do `(None, value) = cursor.get_raw_key_value()`, but I think the suggested change makes this code more robust.

Reply:
The 'standard' way to specify that you want to ignore one or more values in a tuple is to use a throw-away variable name, typically called `_`. See https://stackoverflow.com/questions/5893163/what-is-the-purpose-of-the-single-underscore-variable-in-python

So `(_, value) = cursor.get_raw_key_value()` is valid but still requires that the two values are returned in the tuple, just that one is then ignored.

Reply:
So, I don't think there's a way for a single NULL pointer to be passed into this `typemap` without there being a bug in the code. The `typemap` I created is almost identical to the existing one used by `_get_version_cursor_value`:

`%typemap(argout)(char **metadatap, int *metadatasizep, char **datap, int *datasizep) (
    PyObject *metadata, PyObject *data) {
	if (*$1 && *$3) {
		metadata = PyBytes_FromStringAndSize(*$1, *$2);
		$result = metadata;
		data = PyBytes_FromStringAndSize(*$3, *$4);
		$result = SWIG_Python_AppendOutput($result, data);
	}
}`

Reply:
Thanks for the explanation. It makes sense.

Reply:
To play devil's advocate here, would we consider having an assert to require `*$1` and `*$3` to be set (also in the typemap used by `get_version_cursor_value`).  If it's ever the case the one of them is NULL, that would be surprising, and we might be silently succeeding.

I see we don't use asserts in this code, but we do throw exceptions.

Reply:
BTW, what happens if Python just says: `cursor.get_raw_key_value()` without an assignment to a return variable?  Perhaps that would trigger the error?  Maybe that needs to be in one of the Python tests.

Reply:
Good idea. I've added just such a test (and the call works, even if the result is ignored).